### PR TITLE
[RU]Senkuro and Senkognito new source

### PR DESCRIPTION
**Pull request by e-shl**
Closes https://github.com/tachiyomiorg/tachiyomi-extensions/issues/16987
Closes https://github.com/tachiyomiorg/tachiyomi-extensions/issues/15983
Checklist:

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [x] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [x] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [x] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio

### CODE REVIEW

**Comment by Riztard** [comment]
the lint issue probably from extra white space in new/empty line

File: multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/senkuro/SenkuroQueries.kt [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+@Serializable
+data class FetchChapterPagesVariables(
+    val mangaId: String? = null,
+    val chapterId: String? = null,
+)
+
+val CHAPTERS_PAGES_QUERY: String = buildQuery {
+    """
+        query fetchTachiyomiChapterPages(%mangaId: ID!, %chapterId: ID!) {
+                         mangaTachiyomiChapterPages(mangaId: %mangaId,chapterId: %chapterId) {
```
Comment (alessandrojean) [inline multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/senkuro/SenkuroQueries.kt:180]: Can you fix the indentation for all the query strings?

File: multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/senkuro/Senkuro.kt [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+abstract class Senkuro(
+    override val name: String,
+    final override val baseUrl: String,
+    final override val lang: String,
+) : HttpSource() {
+
+    override val supportsLatest = false
+
+    override fun headersBuilder(): Headers.Builder = Headers.Builder()
+        .add("User-Agent", "Tachiyomi (+https://github.com/tachiyomiorg/tachiyomi)")
```
Comment (alessandrojean) [inline multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/senkuro/Senkuro.kt:38]: Have they explicit asked for this to be sent?

**Review by alessandrojean — CHANGES_REQUESTED** [review]

File: multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/senkuro/Senkuro.kt (hunk 2 above)
Comment (e-shl) [reply multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/senkuro/Senkuro.kt:38]: Yes

File: multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/senkuro/SenkuroQueries.kt [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+    data class FiltersDto(
+        val include: List<String>? = null,
+        val exclude: List<String>? = null,
+    )
+}
+
+val SEARCH_QUERY: String = buildQuery {
+    """
+       query searchTachiyomiManga(
+                                    %query: String,
```
Comment (alessandrojean) [inline multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/senkuro/SenkuroQueries.kt:39]: This still seems to be indented with unnecessary spaces. Fix it to use 4 spaces.

File: multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/senkuro/SenkuroQueries.kt [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+
+@Serializable
+data class FetchDetailsVariables(
+    val mangaId: String? = null,
+)
+
+val DETAILS_QUERY: String = buildQuery {
+    """
+        query fetchTachiyomiManga(%mangaId: ID!) {
+                mangaTachiyomiInfo(mangaId: %mangaId) {
```
Comment (alessandrojean) [inline multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/senkuro/SenkuroQueries.kt:94]: Same here.

File: multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/senkuro/SenkuroQueries.kt [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+                }
+        }
+    """
+}
+
+val CHAPTERS_QUERY: String = buildQuery {
+    """
+        query fetchTachiyomiChapters(%mangaId: ID!) {
+                mangaTachiyomiChapters(mangaId: %mangaId) {
+                                                            message
```
Comment (alessandrojean) [inline multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/senkuro/SenkuroQueries.kt:152]: And here and all other queries.

**Review by alessandrojean — CHANGES_REQUESTED** [review]
Also please change the icon to follow the pattern used by all extensions (**square** with rounded corners). Use the tool linked in the CONTRIBUTING.md guide to generate it.

File: multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/senkuro/SenkuroQueries.kt (hunk 3 above)
Comment (e-shl) [reply multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/senkuro/SenkuroQueries.kt]: Enough. This is just a string representation for further conversion.

**Comment by e-shl** [comment]
> _**square**_ with rounded corners

@alessandrojean subjective stylization. And I already violated the pattern  https://github.com/tachiyomiorg/tachiyomi-extensions/tree/master/src/ru/remanga/res

**Comment by alessandrojean** [comment]
Well, that probably passed without a notice in the review process. It still should be a square with rounded borders, please change it.

**Comment by alessandrojean** [comment]
All sources uses it, so that should be fine to indicate. But will add to the CONTRIBUTING.md guide to make explicit as well.

File: multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/senkuro/SenkuroQueries.kt [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+                     }
+                     cover {
+                            original {
+                                      url
+                            }
+                     }
+             }
+          }
+      }
+
```
Comment (alessandrojean) [inline multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/senkuro/SenkuroQueries.kt:83]:
```suggestion
        query searchTachiyomiManga(
            %query: String,
            %type: MangaTachiyomiSearchTypeFilter,
            %status: MangaTachiyomiSearchStatusFilter,
            %translationStatus: MangaTachiyomiSearchTranslationStatusFilter,
            %genre: MangaTachiyomiSearchGenreFilter,
            %tag: MangaTachiyomiSearchTagFilter,
            %format: MangaTachiyomiSearchGenreFilter,
            %rating: MangaTachiyomiSearchTagFilter,
            %offset: Int,
        ) {
            mangaTachiyomiSearch(
                query:%query,
                type: %type,
                status: %status,
                translationStatus: %translationStatus,
                genre: %genre,
                tag: %tag,
                format: %format,
                rating: %rating,
                offset: %offset,
            ) {
                mangas {
                    id
                    slug
                    originalName {
                        lang
                        content
                    }
                    titles {
                        lang
                        content
                    }
                    alternativeNames {
                        lang
                        content
                    }
                    cover {
                        original {
                            url
                        }
                    }
                }
            }
        }
```

File: multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/senkuro/SenkuroQueries.kt [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+                              }
+                              mainStaff {
+                                          roles
+                                          person {
+                                                  name
+                                          }
+                              }
+                }
+        }
+
```
Comment (alessandrojean) [inline multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/senkuro/SenkuroQueries.kt:146]:
```suggestion
        query fetchTachiyomiManga(%mangaId: ID!) {
            mangaTachiyomiInfo(mangaId: %mangaId) {
                id
                slug
                originalName {
                    lang
                    content
                }
                titles {
                    lang
                    content
                }
                alternativeNames {
                    lang
                    content
                }
                localizations {
                    lang
                    description
                }
                type
                rating
                status
                formats
                genres {
                    slug
                    titles {
                        lang
                        content
                    }
                }
                tags {
                    slug
                    titles {
                        lang
                        content
                    }
                }
                translationStatus
                cover {
                  original {
                      url
                  }
                }
                mainStaff {
                    roles
                    person {
                        name
                    }
                }
            }
        }
```

File: multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/senkuro/SenkuroQueries.kt [hunk 8]
```diff
@@ -0,0 +1,248 @@
...
+                                              updatedAt
+                                  }
+                                  teams {
+                                          id
+                                          slug
+                                          name
+                                  }
+                }
+        }
+
```
Comment (alessandrojean) [inline multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/senkuro/SenkuroQueries.kt:172]:
```suggestion
        query fetchTachiyomiChapters(%mangaId: ID!) {
            mangaTachiyomiChapters(mangaId: %mangaId) {
                message
                chapters {
                    id
                    slug
                    branchId
                    name
                    teamIds
                    number
                    volume
                    updatedAt
                }
                teams {
                    id
                    slug
                    name
                }
            }
        }

```

File: multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/senkuro/SenkuroQueries.kt [hunk 9]
```diff
@@ -0,0 +1,248 @@
...
+                    mangaId: %mangaId,
+                    chapterId: %chapterId
+                  )
+                    {
+                      pages {
+                            url
+                      }
+                    }
+                  }
+
```
Comment (alessandrojean) [inline multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/senkuro/SenkuroQueries.kt:198]:
```suggestion
        query fetchTachiyomiChapterPages(
             %mangaId: ID!,
             %chapterId: ID!
        ) {
            mangaTachiyomiChapterPages(
                mangaId: %mangaId,
                chapterId: %chapterId
            ) {
                pages {
                    url
                }
            }
        }
```

File: multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/senkuro/SenkuroQueries.kt [hunk 10]
```diff
@@ -0,0 +1,248 @@
...
+                                tags {
+                                        id
+                                        slug
+                                        titles {
+                                                lang
+                                                content
+                                        }
+                                }
+                }
+        }
```
Comment (alessandrojean) [inline multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/senkuro/SenkuroQueries.kt:246]:
```suggestion
        query fetchTachiyomiSearchFilters {
            mangaTachiyomiSearchFilters {
                genres {
                    id
                    slug
                    titles {
                        lang
                        content
                    }
                }
                tags {
                    id
                    slug
                    titles {
                        lang
                        content
                    }
                }
            }
        }
```

**Review by alessandrojean — CHANGES_REQUESTED** [review]
Indentation is important to keep a good code readability. It doesn't matter if the source server will strip all the spaces or whatsoever, it needs to be clean for any contributor to be able to read and understand properly the code.

I suggested some changes to fix this and will commit the suggestions to enforce the same indentation.

**Pull request merged**